numpy/core/src/multiarray/buffer.c

```diff
@@ -771,17 +771,6 @@ array_getbuffer(PyObject *obj, Py_buffer *view, int flags)
             goto fail;
         }
     }
-    /*
-     * If a read-only buffer is requested on a read-write array, we return a
-     * read-write buffer, which is dubious behavior. But that's why this call
-     * is guarded by PyArray_ISWRITEABLE rather than (flags &
-     * PyBUF_WRITEABLE).
```

Comment:
This comment still applies, right? Perhaps just needs moving down lower.

Reply:
Yeah, will move it and rephrase slightly. It is not dubious in the sense that I think that is what the buffer protocol documents to happen.